### PR TITLE
Recover latest changes from 1.04+ (Jundiyy)

### PR DESCRIPTION
**Pull request by xezon**
Jundiyy sent a file with the name **Beta 3 Alpha 2** and asked to add it to GitHub. I compared it with **Beta 3** and it revealed this diff. I transferred it to our code by hand. Please check.

### CODE REVIEW

File: Patch104pZH/GameFilesEdited/Data/INI/CommandButton.ini [hunk 1]
```diff
@@ -5592,31 +5594,33 @@ CommandButton Chem_Command_PurchaseScienceTechnicalTraining
...
-  Science           = SCIENCE_RebelAmbush2
+  Science           = Chem_SCIENCE_RebelAmbush2
   ButtonImage       = SUToxAmbsh2
   ButtonBorderType  = UPGRADE ; Identifier for the User as to what kind of button this is
 End
 
 CommandButton Chem_Command_PurchaseScienceRebelAmbush3
   Command           = PURCHASE_SCIENCE
-  Science           = SCIENCE_RebelAmbush3
+  Science           = Chem_SCIENCE_RebelAmbush3
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/CommandButton.ini:5614]: This interferes with the AI's ability to use Ambush.

**Comment by commy2** [comment]
This PR should be split into independent parts, depending on what they actually change. Easier to review. Some of the changes here conflict with other PR's, partly because they do the same or similar things. As for the changes, most seem reasonable as they are either trivial cameo changes, or reverts of changes in 1.04+ that fell out of favour.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/ChemicalGeneral.ini [hunk 2]
```diff
@@ -20971,7 +20971,7 @@ Object Chem_GLAInfantryRebel
 
   UpgradeCameo1 = Chem_Upgrade_GLAAnthraxGamma
   UpgradeCameo2 = Upgrade_InfantryCaptureBuilding
-  UpgradeCameo3 = Upgrade_GLACamouflage
+  ;UpgradeCameo3 = Upgrade_GLACamouflage ; Patch104p @bugfix 05/09/2021 Removed cameo
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/ChemicalGeneral.ini:20974]: Not clear to me that this cameo should be removed. The Toxin Rebel actually benefits from the Camouflage upgrade, despite the fact that the Tox faction cannot purchase it using their own buildings.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini [hunk 3]
```diff
@@ -4845,7 +4845,7 @@ Object Demo_GLATunnelNetwork
   ; Patch104p @bugfix commy2 03/09/2021 Fix missing Demolitions upgrade icon on Base defense.
 
   UpgradeCameo1           = Demo_Upgrade_SuicideBomb
-  ;UpgradeCameo2           = NONE
+  ;UpgradeCameo2           = Upgrade_GLACamoNetting
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini:4848]: Don't see the purpose of this line. The building has no Camo Netting for the Demolutions General.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini [hunk 4]
```diff
@@ -5450,7 +5450,7 @@ Object Demo_GLAStingerSite
 
   UpgradeCameo1          = Upgrade_GLAAPRockets
   UpgradeCameo2          = Demo_Upgrade_SuicideBomb
-  ;UpgradeCameo3          = NONE
+  ;UpgradeCameo3          = Upgrade_GLACamoNetting
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini:5453]: Same as above.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/FactionProp.ini [hunk 5]
```diff
@@ -25,7 +25,8 @@ Object MultiplayerBeacon
   ; *** ENGINEERING parameters ***
   ; Patch104p: Add STRUCTURE to avoid game crash where beacon is selected together with units.
   ; Patch104p: Add IMMOBILE to satisfiy Game Engine expectation that structure is immobile.
-  KindOf = UNATTACKABLE SELECTABLE NO_COLLIDE ALWAYS_VISIBLE ALWAYS_SELECTABLE STRUCTURE IMMOBILE
+  ; Patch104p: Add IMMUNE_TO_CAPTURE
+  KindOf = UNATTACKABLE SELECTABLE NO_COLLIDE ALWAYS_VISIBLE ALWAYS_SELECTABLE STRUCTURE IMMOBILE IMMUNE_TO_CAPTURE
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/FactionProp.ini:29]: There is still debate on this change. See https://github.com/xezon/GeneralsGamePatch/issues/202

File: Patch104pZH/GameFilesEdited/Data/INI/Object/InfantryGeneral.ini [hunk 6]
```diff
@@ -947,7 +947,7 @@ Object Infa_ChinaInfantryTankHunter
   SelectPortrait         = SNTankHunter_L
   ButtonImage            = SNTankHunter
 
-  UpgradeCameo1 = UPGRADE:Fanaticism
+  UpgradeCameo1 = Upgrade_Fanaticism ; Patch104p @bugfix 05/09/2021 Fixed cameo typos
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/InfantryGeneral.ini:950]: I'd argue the unit should sport Nationalism and Fanatism as upgrade icon (even though they share the same art), as it benefits from both, independently. Same for Minigunner.

**Comment by xezon** [comment]
I would prefer if Jundiyy can take over from here. I am not confident that I will not make mistake here.

**Comment by commy2** [comment]
Parts:
- [ ] Demo Rebel Ambush fix
- [ ] Tox Rebel Ambush fix
- [x] Revert Composite tweak for 
- [x] AirF_AmericaTankMicrowave
- [x] AmericaTankCrusader (2)
- [x] AmericaTankPaladin
- [x] AmericaTankMicrowave
- [x] Lazr_AmericaTankCrusader (2)
- [x] SupW_AmericaTankMicrowave
- [x] Demo Rebel Booby trap
- [x] Fix Demo_GLAInfantryTerrorist 1.04+ bug
- [x] Fix SupW_AmericaAirfield cost
- [x] Fanatisicm upgrade icon
- [x] SCUD Launcher AP Rockets upgrade icon
- [x] Paladin $1000

**Comment by commy2** [comment]
Scud Launcher AP Rockets upgrade icon was already fixed in https://github.com/xezon/GeneralsGamePatch/pull/207

**Comment by Jundiyy** [comment]
All good, just need the Paladin Tank change.

**Comment by commy2** [comment]
Close because all done in separate PR's - except Ambush, which is on my task list.

**Pull request closed without merging**